src/components/core/tests/GeoChart.spec.tsx

```diff
@@ -128,6 +131,80 @@ describe("GeoChart", () => {
                 color: "rgb(0,193,141)",
                 isVisible: true,
             },
+            {
+                name: "Speciality",
+                legendIndex: 2,
+                color: "rgb(229,77,66)",
+                isVisible: true,
+            },
+            {
+                name: "Convenience",
+                legendIndex: 3,
+                color: "rgb(241,134,0)",
+                isVisible: true,
+            },
+        ];
+        const wrapper = renderComponent(initProps);
+        wrapper.setProps(props);
+        expect(wrapper.state("enabledLegendItems")).toEqual(expectedState);
+    });
+
+    it("should update GeoChartInner's state with new props with colorMapping", () => {
+        const colorMapping: IColorMapping[] = [
+            {
+                predicate: (headerItem: Execution.IResultAttributeHeaderItem) =>
+                    headerItem.attributeHeaderItem &&
+                    headerItem.attributeHeaderItem.uri === "/gdc/md/storybook/obj/23/elements?id=1", // find attribute item by uri
+                color: {
+                    type: "guid",
+                    value: "2",
+                },
+            },
+        ];
+        const initProps: Partial<IGeoChartInnerProps> = {
+            config: {
+                ...getGeoConfig({ isWithLocation: true, isWithSegment: true, isWithSize: true }),
+                colorMapping,
+            },
+            execution: null,
+        };
+
+        const props: Partial<IGeoChartInnerProps> = {
+            config: {
+                ...getGeoConfig({ isWithLocation: true, isWithSegment: true, isWithSize: true }),
+                colorMapping,
+            },
+            execution: {
+                executionResponse: getExecutionResponse(true, true, false, true),
+                executionResult: getExecutionResult(true, true, false, true),
+            },
+            dataSource: locationSizeColorSegmentFiltersDataSource,
+        };
+        const expectedState = [
+            {
+                name: "General Goods",
+                legendIndex: 0,
+                color: "rgb(0,193,141)",
+                isVisible: true,
+            },
+            {
+                name: "Toy Store",
+                legendIndex: 1,
+                color: "rgb(0,193,141)",
+                isVisible: true,
+            },
+            {
+                name: "Speciality",
+                legendIndex: 2,
+                color: "rgb(229,77,66)",
+                isVisible: true,
+            },
+            {
+                name: "Convenience",
+                legendIndex: 3,
+                color: "rgb(241,134,0)",
+                isVisible: true,
+            },
```

Comment:
We should have one more case for changing color
Steps:
1. render geo component without color in props
2. assert default color in legend item
3. set props with any type of color
4. assert new color in legend item

Reply:
added